### PR TITLE
122/fetch auth modes

### PR DESCRIPTION
**Pull request by swatigogia2020**

### CODE REVIEW

File: gatewayStubConfigurations/imposters/fetchmodesResponse.ejs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+function(config) {
+
+config.logger.info(config.request.body);
+config.logger.info(config.request.path + " is called");
+var reqBody = JSON.parse(config.request.body);
+var reqId = reqBody.requestId;
+var token =  config.request.headers.Authorization;
+config.logger.info("Token: "+token);
```
Comment (praveenadayanand) [inline gatewayStubConfigurations/imposters/fetchmodesResponse.ejs:8]: Should these logging statements  be present?

File: src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            GatewayFetchModesRequestRepresentation gr =
+                new GatewayFetchModesRequestRepresentation(requestId, timeStamp, query);
+            
+            try
+            {
+                logger.LogInformation("{cmSuffix} {correlationId}{healthid}", cmSuffix, correlationId,
+                    fetchRequest.healthId);
+                logger.LogInformation("{gr}",gr.dump(gr));
+                await gatewayClient.SendDataToGateway(PATH_FETCH_AUTH_MODES, gr, cmSuffix, correlationId);
+                //requestmap.add(reqId, [""]); return if the reqid ia lready in the map
```
Comment (praveenadayanand) [inline src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs:54]: Remove commented code

File: src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            DateTime timeStamp = DateTime.Now.ToUniversalTime();
+            Guid requestId = Guid.NewGuid();
+            GatewayFetchModesRequestRepresentation gr =
+                new GatewayFetchModesRequestRepresentation(requestId, timeStamp, query);
+            
+            try
+            {
+                logger.LogInformation("{cmSuffix} {correlationId}{healthid}", cmSuffix, correlationId,
+                    fetchRequest.healthId);
+                logger.LogInformation("{gr}",gr.dump(gr));
```
Comment (praveenadayanand) [inline src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs:52]: Remove extra log statements

File: src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                            "Response about to be send for {RequestId} with {@AuthModes}",
+                            requestId, FetchModeMap.requestIdToFetchMode[requestId]
+                        );
+                        return FetchModeMap.requestIdToFetchMode[requestId];
+                    }
+                    
+                    i++;
+                } while (i < 5);
+
+                //throw new TimeoutException("Timeout for request_id: " + requestId);
```
Comment (praveenadayanand) [inline src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs:72]: Same as above

File: src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            {
+                Log.Information($" Error Code:{request.Error.Code}," +
+                                $" Error Message:{request.Error.Message}.");
+            }
+            else if (request.Auth != null)
+            {
+                string authModes = "";
+                foreach (Mode mode in request.Auth.Modes)
+                {
+                    authModes += mode + ",";
```
Comment (praveenadayanand) [inline src/In.ProjectEKA.HipService/Linkage/FetchModeController.cs:98]: Is there no String.Join method to concatenate enums in C#?

**Review by praveenadayanand — CHANGES_REQUESTED** [review]

File: gatewayStubConfigurations/imposters/fetchmodesResponse.ejs (hunk 1 above)
Comment (swatigogia2020) [reply gatewayStubConfigurations/imposters/fetchmodesResponse.ejs:8]: we can take it has stub refactoring task, can keep it for now to make sure things work on QA and if any debugging required.

File: src/In.ProjectEKA.HipService/Linkage/FetchModeMap.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+using System;
+using System.Collections.Generic;
+
+namespace In.ProjectEKA.HipService.Linkage
+{
+    public class FetchModeMap{
+        public static Dictionary<Guid, string> requestIdToFetchMode = new Dictionary<Guid, string>();
```
Comment (Mahesh-I) [inline src/In.ProjectEKA.HipService/Linkage/FetchModeMap.cs:7]: Can't we have this map in the class itself instead of creating a new class.

**Pull request merged**